### PR TITLE
Developer Guide: Add Implementation Details for Person Roles

### PR DESCRIPTION
**Pull request by purivirakarin**
# Update Developer Guide and Class Diagram for New Person Types

## Overview

This pull request updates the Developer Guide (DG) and the class diagram to reflect the recent integration of new person types (`Client`, `Supplier`, and `Employee`) into our application model. The corresponding implementation has already been merged into the main codebase, as outlined in [PR #50](https://github.com/AY2324S2-CS2103T-F12-1/tp/pull/50). This documentation update ensures that our project's documentation is consistent with the current state of the code, providing clear and up-to-date guidance for developers and contributors.

### Objectives

* **Developer Guide Update**: Revise the DG to include detailed descriptions and implementation details for `Client`, `Supplier`, and `Employee` classes, along with their associated attributes `Department`, `JobTitle`, `Products`, `Skills`, `TermsOfService`.
* **Class Diagram Revision**: Update the project's class diagram to incorporate the new subclasses of `Person` and their relationships with the newly introduced classes.

## Changes to the Developer Guide

1. **Feature Descriptions**: Added comprehensive sections for each new person type, detailing their purpose, associated attributes, and role within the application.
2. **Implementation Details**: Included subsections under each new person type to describe the implementation strategy, focusing on class extensions, attribute encapsulation, and modifications to existing components to support these additions.
3. **Sequence Diagrams**: Introduced new sequence diagrams to visually represent the interactions involving `Client`, `Supplier`, and `Employee` objects within the system.
4. **Usage Scenarios**: Provided example scenarios that illustrate how users might interact with the new functionalities, showcasing the practical applications of these enhancements.
5. **Design Considerations**: Discussed the design choices made during the implementation of new person types, including the rationale behind certain decisions and alternative approaches considered.

## Update to the Class Diagram

* The class diagram has been revised to reflect the extended model hierarchy, showing `Client`, `Supplier`, and `Employee` as subclasses of `Person`.
* New associations between these subclasses and their specific attributes (`Department`, `JobTitle`, `Products`, `Skills`, `TermsOfService`) have been added to the diagram.
* Adjustments have been made to ensure clarity and coherence with the updated model structure, facilitating a better understanding of the system's architecture.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -159,6 +159,50 @@ Classes used by multiple components are in the `seedu.netconnect.commons` packag
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Person Roles (Employee, Client, and Supplier)
+
+This feature update introduces three new person types into the existing Person model: `Client`, `Supplier`, and `Employee`. These classes extend the base `Person` class and encapsulate various role-specific functionalities and attributes, improving the application's ability to cater to a diverse range of user interactions.
```
Comment (tanjieling0) [inline docs/DeveloperGuide.md]: Should we be using terms like 'new' im not sure about this.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -159,6 +159,50 @@ Classes used by multiple components are in the `seedu.netconnect.commons` packag
...
+#### Implementation
+
+##### Class Additions
+
+* `Client`, `Supplier`, and `Employee` classes have been added, extending the `Person` class to include role-specific fields.
+* New classes `Department`, `JobTitle`, `Products`, `Skills`, and `TermsOfService` are introduced to encapsulate relevant attributes.
+
+##### Data Storage
+
+* The `JsonAdaptedPerson` class has been updated to support the conversion to and from the new role-based classes, ensuring compatibility with the enhanced JSON schema.
```
Comment (tanjieling0) [inline docs/DeveloperGuide.md]: similar to above, not sure about the word 'updated'

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (purivirakarin) [reply docs/DeveloperGuide.md]: Updated!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (purivirakarin) [reply docs/DeveloperGuide.md]: Updated!

**Review by tanjieling0 — APPROVED** [review]

**Pull request merged**